Use Django migrations (1.5 release)

Hwesta opened the pull request:
Use Django's migrations, instead of custom-written SQL for database changes. 
- Add schema migrations for all changes since 1.4.1
- Add data migrations for changes since 1.4.1
- Use dumpdata fixture for initial data
- Update code to reflect data - eg. set auto_now on lastModified columns, update Event creation to actually have unique IDs
- Includes #395/#417 as a migration
- Removes old SQL migrations
- Includes artefactual/archivematica-fpr-admin#14
- Edit: Update version & adds version update migration

Todo:
- [ ] Use data migration instead of loaddata for initial data migration
- [ ] Investigate why initial schema migration is slow
- [x] Test with upgrades
- [x] Test with artefactual-labs/ansible-role-archivematica-src#34

Hwesta committed:
Migrations: Remove old SQL migrations

Hwesta committed:
Migrations: Add FPR migrations

Hwesta committed:
Migrations: Add migrations for 1.4.1 to 1.5

Add table migrations for 1.4.1 to 1.5. Update dev-helper to run migrations
instead of SQL file.

Hwesta committed:
Migrations: data vs schema fixes

Set timezone, delete dead entries, set auto_now on columns.

Hwesta committed:
Migrations: Data migrations for 1.4.1 to 1.5

Hwesta committed:
Migrations: Store DIP

Migrations port of ea425064f2c90bdc4d2c4079621d16fb289eb0ef

Store DIP after upload. refs #8995

Added, to the microservice chain, logic to ask the user if they'd
like to store the DIP after upload.

Hwesta committed:
Migrations: Event ID is unique, update code

Since Event.event_id is the destination of a foreign key, it must be unique.
Update code that creates events with non-unique IDs.

mistydemeo commented on src/dashboard/src/requirements/base.txt:
```diff
@@ -11,6 +11,7 @@ elasticsearch>=1.0.0,<2.0.0
 lazy-paged-sequence
 # Required by storage-service component
 slumber==0.6.0
+sqlparse
```
What's this used by?

Hwesta replied:
I think I added this when I was using RunSQL instead of RunPython for the data migrations. It can probably be removed.

mistydemeo commented:
:raised_hands:

Hwesta commented on src/dashboard/src/main/migrations/0010_dip_upload_store.py:
```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```
This should be checked to make sure it matches #417, which is what was actually merged.

Hwesta replied:
Checked and updated.

Hwesta commented on src/dashboard/src/main/migrations/0008_fpcommandoutput.py:
```diff
@@ -0,0 +1,18 @@
+        migrations.AlterModelTable(
+            name='fpcommandoutput',
+            table='FPCommandOutput',
+        ),
```
This needs to be updated to handle legacy databases.

Context: We originally had a table named FPCommandOutput.  When we added models, a FPCommandOutput model was created, but no legacy table name was created, so Django created main_fpcommandoutput.  This was later fixed (ab97a22203031bfca8e2bfa8906b18e9a62c2857) and this migration is intended to move main_fpcommandoutput to FPCommandOutput, but it fails if the legacy FPCommandOutput already exists.  We need to establish which table actually contains user data, and preserve that when we merged the two tables.

Hwesta replied:
Updated this to revert the change pointing FPCommandOutput model at FPCommandOutput table. Modified this migration to move all data from FPCommandOutput table into main_fpcommandoutput table.

Hwesta committed:
Migrations: Fix FPCommandOutput table

Move FPCommandOutput content into main_fpcommandoutput. Point the
FPCommandOutput model at main_fpcommandoutput.

refs #9735

Hwesta committed:
Update Archivematica version

Hwesta committed:
Migrations: Initial data migration

sevein commented:
:shipit:

Hwesta commented on src/dashboard/src/create_datamigration.py:
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
```
This should be moved to devtools.

Hwesta commented on localDevSetup/recreateDB.sh:
```diff
@@ -57,19 +57,13 @@ echo "Creating MCP database"
 echo "Creating ${username} user"
 mysql -u root "${dbpassword}" --execute="CREATE USER '${username}'@'localhost' IDENTIFIED BY '${password}';"
-mysql -u root "${dbpassword}" --execute="GRANT SELECT, UPDATE, INSERT, DELETE, CREATE, ALTER, INDEX ON ${databaseName}.* TO '${username}'@'localhost';"
+mysql -u root "${dbpassword}" --execute="GRANT SELECT, UPDATE, INSERT, DELETE, CREATE, ALTER, INDEX, DROP ON ${databaseName}.* TO '${username}'@'localhost';"
```
This is no longer necessary.
